core/common/buffer.hpp

```diff
@@ -222,6 +222,22 @@ namespace kagome::common {
     return s << buffer.toVector();
   }
 
+  /**
+   * @brief decodes buffer object from stream
+   * @tparam Stream input stream type
+   * @param s stream reference
+   * @param buffer value to decode
+   * @return reference to stream
+   */
+  template <class Stream>
+  Stream &operator>>(Stream &s, Buffer &buffer) {
+    std::vector<uint8_t> data;
+    s >> data;
+    buffer.clear();
```

Comment:
why clear?
what if I do 

```C++
Stream s;
Buffer a;
Buffer b;
s >> a >> b; // ?
```

Reply:
When we decode an item we don't suggest the value appended to variable content.
Buffer is a sequence of bytes decoded as collection. l don't think Buffer should keep old data when new data is read and stored to it.

Reply:
if we succeed receiving data from stream, only then clear the buffer.